dora/core/client/fs/src/main/java/alluxio/client/file/dora/ConsistentHashPolicy.java

```diff
@@ -16,19 +16,27 @@
 import alluxio.conf.AlluxioConfiguration;
 import alluxio.conf.PropertyKey;
 import alluxio.exception.status.ResourceExhaustedException;
+import alluxio.wire.WorkerIdentity;
+
+import com.google.common.collect.ImmutableList;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 import java.util.List;
+import java.util.stream.Collectors;
 
 /**
  * An implementation of WorkerLocationPolicy.
- *
+ * <p>
  * A policy where a file path is matched to worker(s) by a consistenct hashing algorithm.
  * The hash algorithm makes sure the same path maps to the same worker sequence.
  * On top of that, consistent hashing makes sure worker membership changes incur minimal
  * hash changes.
  */
 public class ConsistentHashPolicy implements WorkerLocationPolicy {
-  private static final ConsistentHashProvider HASH_PROVIDER =
+  private static final Logger LOG = LoggerFactory.getLogger(ConsistentHashPolicy.class);
+
+  private final ConsistentHashProvider mHashProvider =
```

Comment:
@jiacheliu3 do you remember why `HASH_PROVIDER` was a static field? The static-ness and its internal cache of worker lists are causing integration tests to fail, because workers generated in last test is carried over to next test.

Reply:
https://github.com/Alluxio/alluxio/pull/17753#discussion_r1258440617
I believe the intention was to make this a process-scope singleton